src/mock.js

```diff
@@ -82,6 +82,21 @@ function addMockFunctionsToSchema({ schema, mocks = {}, preserveResolvers = fals
     return customMock;
   }
 
+  function assignResolveType(type) {
+    const fieldType = getNullableType(type);
+    const namedFieldType = getNamedType(fieldType);
+
+    if (namedFieldType instanceof GraphQLUnionType ||
+        namedFieldType instanceof GraphQLInterfaceType
+    ) {
+      const oldResolveType = namedFieldType.resolveType;
+      // the default `resolveType` always returns null. We add a fallback
+      // resolution that works with how unions and interface are mocked
+      namedFieldType.resolveType = (data, context, info) =>
+        oldResolveType(data, context, info) || info.schema.getType(data.typename);
```

Comment:
I don't understand your comment tho... if `preserveResolvers` is false this works, because `resolveType` returns null and it fallbacks to the `info.schema.getType(data.typename)` part